src/agent.cpp

```diff
@@ -249,83 +260,89 @@ void Agent::actionImpl() {
   }
 
   // For now let's not worry about turning the neck or setting the vision.
-  // However, do this now so doesn't override anything changed by the requested action.
-  // TODO for librcsc: setViewActionDefault, setNeckActionDefault that will not overwrite if already set.
+  // But do the settings now, so that doesn't override any set by the actions below.
+  // TODO: Add setViewActionDefault, setNeckActionDefault to librcsc that only set if not already set.
   this->setViewAction(new View_Tactical());
   this->setNeckAction(new Neck_TurnToBallOrScan());
 
+  const WorldModel & wm = this->world();
+  bool may_fix = wm.self().collidesWithPost();
+
   switch(requested_action) {
     case DASH:
-      this->doDash(params[0], params[1]);
+      setLastActionStatusCollision(may_fix, this->doDash(params[0], params[1]));
```

Comment:
last_action_status = this->doDash(...);
last_action_status |= wm.self().collidesWithPost(); // Dashing can be effective when collided with post
break;